### ISSUE TITLE
spaced passwords add too much entropy.

### ISSUE THREAD

**Issue by gabeio** [dropbox/zxcvbn#21]
a b c d e f g h i j k l m n o p q r s t u v w x y z 0 1 2 3 4 5 6 7 8 9
entropy:417.915
crack_time(seconds):3.190943210320069e+121
crack_time(readable):centuries
score:4

abcdefghijklmnopqrstuvwxyz0123456789
entropy:14.118
crack_time(seconds):0.889
crack_time(readable):instant
score:0

a b c 1 2 3
entropy:59.299
crack_time(seconds):35452087835576.43
crack_time(readable):centuries
score:4

entropy:3.807
crack_time(seconds):0.001
crack_time(readable):instant
score:0

just adding a space between every letter does not make it all that much more secure...

**Comment by Terebi42** [comment]
correct horse battery staple has entropy of 62.86
correcthorsebatterystaple has 45. 

Since diceware and makemeapassword both make spaces the default now, and this is likely the most common "secure" pattern, the wordwordwordword guess should assume spaces and not grant them any entropy.

**Comment by hickford** [comment]
> zxcvbn('passwordqwerty').guesses_log10
    4.176091259055681
    > zxcvbn('password qwerty').guesses_log10
    8.000715995361274

**Comment by lowe** [comment]
yup, this is on the list of things to fix. if you have suggestions, let me know. 

one idea is to run zxcvbn several times according to the following substitutions, taking the minimum, and adding an extra bit or two for the choice of delimiter:

```
original
original, s/ //g
original, s/-//g
original s/_//g
original s/,//g
original s/~//g
```

this grows the runtime 6x in the worst case, and isn't a general solution. but it has the advantage of simplicity, and zxcvbn is currently very fast. it also wouldn't be 6x in the typical case -- it could be coded to only check a substitution when the delimiter char appears in the string.

another idea would be to use a regex to try to detect the delimiter automatically, instead of having a hardcoded list. something along the lines of: 
http://stackoverflow.com/questions/16333681/regex-repeated-words-on-the-same-line

open to more ideas.
